src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -158,6 +163,24 @@
             Setting.Property.NodeScope,
             Setting.Property.Dynamic);
 
+    public static final Setting<Long> MODEL_CACHE_SIZE_IN_BYTES_SETTING = new Setting<>(MODEL_CACHE_SIZE_IN_BYTES,
+            (s) -> Long.toString(KNN_DEFAULT_MODEL_CACHE_SIZE_IN_BYTES),
+            (s) -> {
+                    long value = Long.parseLong(s);
+                    if (value < KNN_MIN_MODEL_CACHE_SIZE_IN_BYTES) {
```

Comment:
nit: why not combine both into single check and error message as  "value must be > 100 KB and <= 80 MB, so that user knows required setting at first failure attempt?

Reply:
Sure will update